### DISCUSSION TITLE
Welcome to Stacks Discussions!

### DISCUSSION THREAD

**Discussion by cablehead** [cablehead/stacks#36] [Announcements]
## 👋 Welcome!

  We’re using Discussions as a place to connect with other Stacks' users. We hope that you:

  * Ask questions you’re wondering about.
  * Share ideas and make feature requests.
  * Connect with fellow community members to explore creative workflows and innovative combinations of tools.
  * Welcome others and are open-minded. Remember that this is a community we  build together 💪.

  To get started, comment below with your experiences using Stacks.

If you'd prefer a more real-time medium, Stacks also has a [discord channel](https://discord.com/invite/YNbScHBHrh).

**Comment by pranavmyname** [comment]
Hey, how can I set the maximum number of days, the app should be recording the clips for? I don't see any options to access the settings.

**Comment by cablehead** [comment]
Hi! There's currently no limit to the amount of history that will be stored (and no option to cap it).

There's a bunch of features I'm hoping to support in the "tools of thought" space, that will benefit from being able to retain years of history, for the clips you care about.

Do you mind if I ask your use case for setting a retention limit?

Performance wise, I'm a heavy user and regularly capture images and screenshots. My personal store now has 6 months of history, and most things are still very snappy. There's a lot of levers left to pull to improve performance more, as well.

For disk usage, my personal store is at 420MB, again after 6 months with a lot of images.

My current thinking for capping retention is: you set an upper bound on the amount of storage you want to use, and stacks will eject older clips: prioritizing clips which are old, haven't been touched again, and large.

What are your thoughts?

**Reply by pranavmyname** [reply]
Hi, my use case is purely for the disk usage limit. I use a macbook with 500GB storage and have to regularly clean up data to work with newer clients. I don't know how much space it is using up right now and I was concerned if it should be allowed to use the disk space without any limits especially since I capture multiple screenshots and work with a lot of image data on day-to-day basis. I used copy less 2 and it stored data in a queue (I get it) where data older than say 7 days get pushed out.

**Reply by cablehead** [reply]
Gotcha, appreciate the feedback. If I add the ability to cap storage, would you prefer it to be timed based (number of days), or size base (number of Mbs)?
